builtin/docker/builder.go

```diff
@@ -154,6 +157,15 @@ func (b *Builder) Build(
 	}
 	cli.NegotiateAPIVersion(ctx)
 
+	buildArgs := make(map[string]*string)
+
+	if b.config.BuildArgs != nil {
+		for _, arg := range b.config.BuildArgs {
+			pair := strings.Split(arg, "=")
+			buildArgs[pair[0]] = &pair[1]
```

Comment:
It looks like we build up `buildArgs` here, but later only use `b.config.BuildArgs`? Maybe we could move this right above where `buildArgs` is used since we're only needing it conditionally later on after all the fallback if-statements.

Reply:
@briancain, you mean it 

`

    } else {
		buildArgs := make(map[string]*string)
		if b.config.BuildArgs != nil {
			for _, arg := range b.config.BuildArgs {
				pair := strings.Split(arg, "=")
				buildArgs[pair[0]] = &pair[1]
			}
		}

		// No fallback, build with Docker
		step.Done()
		step = nil
		if err := b.buildWithDocker(
			ctx, ui, sg, cli, contextDir, relDockerfile, result.Name(), buildArgs,
		); err != nil {
			return nil, err
		}
	}
`